pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -67,9 +67,77 @@ type BindMechanism interface {
 	startDHCP() error
 }
 
-type podNICImpl struct {
-	cacheFactory cache.InterfaceCacheFactory
-	handler      NetworkHandler
+type podNIC struct {
+	vmi              *v1.VirtualMachineInstance
+	podInterfaceName string
+	launcherPID      *int
+	iface            *v1.Interface
```

Comment:
Why a pointer ? Do you plan to mutate it somehow ? I don't think we can go anywhere without one.

Reply:
Meh, I guess the same could be said about the vmi, and the network.

Can be done in a follow-up.

Reply:
Meh, I guess the same could be said about the vmi, and the network.

Can be done in a follow-up.